### PR TITLE
Fix topAdjusted sort

### PR DESCRIPTION
**Pull request by oetherington**
Another airport fix.

This one fixes sorting by `topAdjusted` (and some other minor bugs too). It includes:

- Implementing `$ifNull`
- Better handling of `$arrayElemAt` (this is still a bit janky and can almost certainly be further improved in the future, but for now I just want to kill this bug ASAP)
- Making synthetic field names resolvable
- A better index for `topAdjusted` (still isn't perfect, but it's better than what we were using before)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203634886862408) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
Assigning @jpaddison3 because you're the only dev I know is definitely working this week - feel free to reassign to someone else if you're too busy!

File: packages/lesswrong/lib/sql/Query.ts [hunk 1]
```diff
@@ -58,6 +58,12 @@ const arithmeticOps = {
   ...comparisonOps,
 };
 
+const variadicFunctions = {
+  $min: "LEAST",
+  $max: "GREATEST",
+  $ifNull: "COALESCE",
+};
```
Comment (jpaddison3) [inline packages/lesswrong/lib/sql/Query.ts:65]:
```suggestion
} as const;
```

Yeah, I know, just doing that won't actually make the type of `func` below any better.

File: packages/lesswrong/lib/sql/Query.ts [hunk 2]
```diff
@@ -526,19 +532,24 @@ abstract class Query<T extends DbObject> {
...
       const [array, index] = expr[op];
-      if (typeof array !== "string" || array[0] !== "$" || typeof index !== "number") {
-        throw new Error("Invalid arguments to $arrayElemAt");
-      }
-      const tokens = array.split(".");
-      const field = tokens[0][0] === "$" ? tokens[0].slice(1) : tokens[0];
-      const path = tokens.slice(1).flatMap((name) => ["->", `'${name}'`]);
-      if (path.length) {
-        path[path.length - 2] = "->>";
+      let field;
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/lib/sql/Query.ts:537]: Is that right?

**Review by jpaddison3 — COMMENTED** [review]
I'm at like 60% that I've found a bug. Would somewhat appreciate reviewing again. But I know you're sneaking in fixes here and there so no worries if you want to #shipit without waiting for me.

If you don't have time to get it, I'll hopefully have a chance to take it from here tomorrow.

**Review by jpaddison3 — APPROVED** [review]
LGTM

**Pull request merged**